scripts/typescript/src/pppdb/pppdb.ts

```diff
@@ -0,0 +1,257 @@
+    //
+
+    if( data.reference ) {
+      const lower_case_doi_or_isbn = data.reference.toLocaleLowerCase();
+
+      if ( lower_case_doi_or_isbn.startsWith('doi:')) {
+        result = {
+          node: ['Reference'],
+          title: '',
+          doi: lower_case_doi_or_isbn.replace('doi:', '').trim(),
+          type: 'journal_article'
+        }
+      } else if (lower_case_doi_or_isbn.startsWith('isbn:')) {
+        result = {
+          node: ['Reference'],
+          title: '',
+          doi: lower_case_doi_or_isbn.replace('isbn:', '').trim(), // TODO: ISBN should NOT be stored as a DOI. We need to identify where we should store it.
+          type: 'book'
+        }
+      } else {
+        this.debug(`Unknown reference: ${data.reference}. Only handle 'doi:' and 'isbn:' for now.`);
+      }
+    } else if ( data.doi ) {
+      result = {
+        node: ['Reference'],
+        title: '',
+        doi: data.doi.toLocaleLowerCase().trim(),
+        type: 'journal_article'
+      }
+    } else {
+      this.debug(`Cannot find any reference or doi. This case is not implemented yet.`)
+    }
+
+    return result;
+  }
+
+  /**
+   * Loads to first sheet of a given *.xslx file
+   * 
+   * User has to know in advance the type of the expected JSON,
+   * No check is done prior to return the data. In case you want to ensure the data has the right type
+   * you must not set T and do discrimination tests.
+   */
+  private loadSheet<T extends { row: number }>(file_path: string, options?: XLSX.ParsingOptions) {
+
+    // Load the XLSX file
+    process.stdout.write(`-- Reading ${file_path} ...\n`);
+    const workBook = XLSX.readFile(file_path, options);
+
+    // Pick the first sheetName
+    const sheetName = workBook.SheetNames.at(0);
+    if( !sheetName ) throw new Error(`Unable to find any sheets in ${file_path}`)
+
+    // Loading the sheet
+    process.stdout.write(`-- Getting '${sheetName}' WorkSheet ...\n`);
+    const workSheet: XLSX.WorkSheet = workBook.Sheets[sheetName] ?? [];
+  
+    // Convert to objects
+    //   Note: We'll loose in performance and memory usage by doing this, but our dataset is very small.
+    //         We prefer here to deal with objects instead of defining a Column enum (cf. rcbc project).
+    process.stdout.write(`-- Converting WorkSheet to JSON ...\n`);
+    const result = XLSX.utils.sheet_to_json<T>(workSheet);
+    // store the row index to help debugging.
+    result.forEach( (each, index) => each.row = index + 1 ); // +1 for the header
+
+    process.stdout.write(`-- Found ${result.length} row(s)\n`);
+    process.stdout.write(`-- Here is a sample of the first row:\n ${JSON.stringify(result.slice(undefined, 1), null, '\t')}\n`);
+
+    return result;
+  }
+  
+  /**
+   * Simple function to log a warning related to a given *.xslx row index 
+   */
+  private debug(...args: any[]) {
+    console.debug(`-- row ${this.currentRow}: `, ...args);
+  }
+
+  private addMaterial(material: IMaterial) {
```

Comment:
```suggestion

 /**
 * Add a given material to the project, and checks if the material's name is unique.
 * In case the name is in use, a log is emitted.
 */
  private addMaterial(material: IMaterial) {
```